tiledb/bioimg/converters/base.py

```diff
@@ -252,6 +270,60 @@ def tile_to_tiledb(level_tile: Tuple[slice, ...]) -> None:
                     else:
                         a[:] = axes_mapper.map_array(reader.level_image(level))
 
+                    if generate_pyramid:
+                        scaler = Scaler(
+                            a,
+                            Axes("".join(dim.name for dim in a.domain)),
+                            scale_factor=pyramid_scale,
+                        )
+                uris.append(uri)
+
+            if generate_pyramid:
+                level = level_min + 1
+
+                for dimension in scaler.resolutions:
```

Comment:
The body of this loop should be:
(a) extracted to a private method; this method is already long/complex enough without all the pyramid generation logic
(b) refactored to eliminate all duplicated logic (e.g. between the `NON_PROGRESSIVE` vs `CHUNKED_NON_PROGRESSIVE` )